lib/seahorn/PredicateAbstraction.cc

```diff
@@ -63,22 +66,72 @@ namespace seahorn
 		boost::tribool result = fp.query ();
 //
 		if (result) outs () << "sat";
-		else if (!result) outs () << "unsat";
+		else if (!result)
+		{
+			outs() << "unsat\n";
+			HornDbModel absModel(new_db, fp);
+			converter.convert(absModel, oldModel, newToOldPredMap);
+			for(ExprMap::iterator it = oldModel.getRelToSolutionMap().begin(); it!=oldModel.getRelToSolutionMap().end(); ++it)
+			{
+				outs() << *(bind::fname(it->first)) << ": " << *(it->second) << "\n";
+			}
+		}
 		else outs () << "unknown";
 		outs () << "\n";
-//
-		printInvars (M);
 
 		return false;
 	}
 
+	bool HornModelConverter::convert (HornDbModel &in, HornDbModel &out, std::map<Expr, Expr> &newToOldPredMap)
+	{
+		ExprMap inRelToSolMap = in.getRelToSolutionMap();
```

Comment:
Don't expose solution map. Don't make assumptions on how `HornDbModel` is implemented. I am commenting on this line, but the comment applies throughout the code.